### PR TITLE
Tests and basic implementation of Responses for Trace API

### PR DESCRIPTION
**Pull request by b1bart**
In this PR for review:
- [ ] basic response variant formatting w/o ABI considerations
- [ ] basic tests framework for "response" black box
- [ ] basic tests
- [ ] edge test for missing data/metadata
- [ ] edge test for corrupt data/metadata
- [ ] test for deadline exceptions

### CODE REVIEW

File: plugins/trace_api_plugin/include/eosio/trace_api_plugin/request_handler.hpp [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+         // scan for the block offset and latest LIB
+         try {
+            logfile_provider.scan_metadata_log_from(block_height, 0, [&](const metadata_log_entry& e) -> bool {
+               if (e.contains<block_entry_v0>()) {
+                  const auto& block = e.get<block_entry_v0>();
+                  if (block.number == block_height) {
+                     block_offset = block.offset;
+                  }
+               } else if (e.contains<lib_entry_v0>()) {
```
Comment (brianjohnson5972) [inline plugins/trace_api_plugin/include/eosio/trace_api_plugin/request_handler.hpp]: I am not sure how scan_metadata_log_from would return a lib_entry_v0 based on just a block_height value?  I was thinking the lib_entry_v0 was just for the readers understanding of the file data, but now thinking we need to work through the use case that causes the reader to return it.

Comment (b1bart) [reply plugins/trace_api_plugin/include/eosio/trace_api_plugin/request_handler.hpp]: The logfile reader should just be reading sequentially with no inspection on what its passing.   As long as the slice file for a given block height contains all `lib_entry_v0` structures relevant to blocks in that slice then we should be able to find the "Best" one from that scan and it should ideally be an entry which is at or past the end of the slice once that slice is final.

Comment (b1bart) [reply plugins/trace_api_plugin/include/eosio/trace_api_plugin/request_handler.hpp]: Will accept the PR and move this Discussion off comments

Comment (brianjohnson5972) [reply plugins/trace_api_plugin/include/eosio/trace_api_plugin/request_handler.hpp]: ignore me, forgot that my reader just keeps reading till you tell it to stop.

**Pull request merged**